search-app/src/cmr/search/services/query_service.clj

```diff
@@ -263,6 +265,21 @@
         results (qe/execute-query context query)]
     (common-search/search-results->response context query results)))
 
+(defn new-granules->collection-ids
+  "Finds granules that were added after a given date and return their parent collection ids.
+   Supports CMR Harvesting."
+  [context params]
+  (when-let [[start-date end-date] (mapv time-format/parse
+                                         (string/split (:has-granules-created-at params) #","))]
+    (let [query (qm/query {:concept-type :granule
+                           :condition (qm/date-range-condition
+                                       :created-at start-date end-date)
+                           :page-size :unlimited
+                           :result-format :query-specified
+                           :result-fields [:collection-concept-id]})
+          new-granules (:items (qe/execute-query context query))]
+      (map :collection-concept-id new-granules))))
```

Comment:
I'm not sure what you are doing with the return value for this function later, but you might want to remove duplicates here before returning this.